scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/Sm4Encrypt.scala

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.ppml.crypto
+
+import com.intel.analytics.bigdl.dllib.utils.Log4Error
+
+import java.io._
+import java.nio.file.{Files, Paths}
+import java.security.SecureRandom
+import java.time.Instant
+import java.util.Arrays
+import javax.crypto.spec.{IvParameterSpec, SecretKeySpec}
+import javax.crypto.{Cipher, Mac}
+import org.apache.spark.input.PortableDataStream
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+
+class SM4Encrypt extends Crypto {
+
+  def encryptFile(binaryFilePath: String, savePath: String, dataKeyPlaintext: String) = {
```

Comment:
for each method, we'd better add unit tests for it.
please refer to https://github.com/intel-analytics/BigDL/blob/main/scala/ppml/src/test/scala/com/intel/analytics/bigdl/ppml/crypto/dataframe/EncryptDataFrameSpec.scala